### ISSUE TITLE
You must implement getStoragePath method for deleting uploaded files.

### ISSUE THREAD

**Issue by FaridAghili** [laravel/nova-issues#2181]
- Laravel Version: 6.6.0
- Nova Version: 2.8.0
- PHP Version: 7.3.0
- Database Driver & Version: MySQL 5.7
- Operating System and Version: Ubuntu 18.08
- Browser type and version: Chrome Desktop 78.0.3904.108

### Description:
When I try to remove a record which have a Trix field (which this Trix field contains a file), or when trying to remove a file (uploaded image, for example) from a Trix field, Nova gives Server error and log file contains:
```
You must implement getStoragePath method for deleting uploaded files.
```

### Steps To Reproduce:
Create a Nova resource with a Trix field, drag and drop a file inside the trix filed, save the model, now try to remove the model or delete the file uploaded into Trix field:
```php
Trix::make('متن خبر', 'content')
    ->withFiles('liara')
    ->prunable()
    ->stacked()
    ->rules('nullable', 'string'),
```

### Stacktrace
```
/var/www/html/nova/src/DeleteField.php:26
/var/www/html/nova/src/Http/Controllers/DeletesFields.php:42
/var/www/html/vendor/laravel/framework/src/Illuminate/Support/Traits/EnumeratesValues.php:176
/var/www/html/nova/src/Http/Controllers/DeletesFields.php:43
/var/www/html/nova/src/Http/Controllers/ResourceDestroyController.php:25
```

### Telescope
```
67     }
68 
69     /**
70      * Get the full path that the field is stored at on disk.
71      *
72      * @return string|null
73      */
74     public function getStoragePath()
75     {
76         throw new RuntimeException('You must implement getStoragePath method for deleting uploaded files.');
77     }
78 }
79 
```

**Comment by jbrooksuk** [comment]
Ping @dkulyk.

Could you perhaps write some documentation around this, please?

**Comment by dkulyk** [comment]
Working on it.

**Comment by FaridAghili** [comment]
@dkulyk 
Is it possible to write changes you made in https://github.com/laravel/nova/pull/740 in this issue?

Because I don't have access to that pull request, as GitHub blocked Iranian's access to private repositories, and also I'm not able to delete some posts from one of my production projects because of this error.

![image](https://user-images.githubusercontent.com/8607021/69917241-168a9280-1479-11ea-9251-1e36751dc9c5.png)

**Comment by dkulyk** [comment]
@FaridAghili src/Fields/Storable.php in last method you should return null instead throw exception.

**Comment by nuril91** [comment]
@FaridAghili try to use SoftDelete in your model..

**Comment by vincenzoraco** [comment]
Did someone look into this issue? The PR https://github.com/laravel/nova/pull/740 failed some checks but it seems nobody replied

**Comment by pedmindset** [comment]
Having the same issue

**Comment by SadeghPM** [comment]
Any progress?

**Comment by sutcliffe121** [comment]
Having the same issue. Any updates on a fix?

"You must implement getStoragePath method for deleting uploaded files."
Message displayed whenever a user tries to delete an entry which uses a Trix field.

**Comment by oriolfg** [comment]
I have exactly the same problem, hope for a solution soon.

  We have purchased a license to test and evaluate the possibility of using NOVA in our projects, but having this problem will be difficult.

**Comment by Milkhan** [comment]
Any progress?

**Comment by Milkhan** [comment]
@dkulyk  storagePath function is currently only throws an error. 
I have updated it as this (nova/src/Fields/Storable.php):
` public function getStoragePath()`
`
    {`
       ` return $this->storagePath;`      
`// throw new RuntimeException('You must implement getStoragePath method for deleting uploaded files.');`
    `}`

And it works now without an error.

**Comment by davidhemphill** [comment]
Fixed via laravel/nova#813
